### PR TITLE
GET Monitoring Security Policy Device List

### PR DESCRIPTION
**Pull request by rtmooy**
# Pull Request summary:
Add support for monitoring security policy device list endpoint

# Description of changes:
Add support for monitoring security policy device list endpoint
New unit test for monitoring security policy device list

# Unit Test:
```
(vtest3.8) tester@SJC24-TBSEC-85-121:tests$ python3.8 test_monitoring_security_policy.py
.
----------------------------------------------------------------------
Ran 1 test in 2.211s

OK
```

```
>>> device_list = session.endpoints.monitoring_security_policy.get_device_list()
>>> device_list
SecurityPolicyDeviceList(amp_down=[], amp_up=['172.16.255.16'], ips_down=[], ips_up=['172.16.255.16'], urlf_down=[], urlf_up=['172.16.255.16'], zbfw_down=[], zbfw_up=[])
>>> device_list.
device_list.amp_down                  device_list.json(                     device_list.model_fields              device_list.model_validate_strings(   device_list.urlf_up
device_list.amp_up                    device_list.model_computed_fields     device_list.model_fields_set          device_list.parse_file(               device_list.validate(
device_list.construct(                device_list.model_config              device_list.model_json_schema(        device_list.parse_obj(                device_list.zbfw_down
device_list.copy(                     device_list.model_construct(          device_list.model_parametrized_name(  device_list.parse_raw(                device_list.zbfw_up
device_list.dict(                     device_list.model_copy(               device_list.model_post_init(          device_list.schema(                   
device_list.from_orm(                 device_list.model_dump(               device_list.model_rebuild(            device_list.schema_json(              
device_list.ips_down                  device_list.model_dump_json(          device_list.model_validate(           device_list.update_forward_refs(      
device_list.ips_up                    device_list.model_extra               device_list.model_validate_json(      device_list.urlf_down                 
>>> device_list.amp_down
[]
>>> device_list.amp_up
['172.16.255.16']
>>> device_list.ips_down
[]
>>> device_list.ips_up
['172.16.255.16']
>>> device_list.urlf_down
[]
>>> device_list.urlf_up
['172.16.255.16']
>>> device_list.zbfw_down
[]
>>> device_list.zbfw_up
[]
```

# Checklist:
- [x] Make sure to run pre-commit before committing changes
- [x] Make sure all checks have passed
- [x] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

### CODE REVIEW

File: catalystwan/models/monitoring/security_policy.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2024 Cisco Systems, Inc. and its affiliates
+
+from pydantic import BaseModel, Field
+
+
+class DeviceListResponse(BaseModel):
```
Comment (sbasan) [inline catalystwan/models/monitoring/security_policy.py:6]: why not to model this in regular way?
image
```suggestion
class SecurityPolicyDeviceList(BaseModel):
    amp_down: List[str]
    amp_up: List[str]
    ips_down: List[str]
    ips_up: List[str]
    urlf_down: List[str]
    urlf_up: List[str]
    zbfw_down: List[str]
    zbfw_up: List[str]
```

File: catalystwan/endpoints/monitoring/security_policy.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2024 Cisco Systems, Inc. and its affiliates
+
+# mypy: disable-error-code="empty-body"
+
+from catalystwan.endpoints import APIEndpoints, get
+from catalystwan.models.monitoring.security_policy import DeviceListResponse
+
+
+class MonitoringSecurityPolicy(APIEndpoints):
+    @get("/security/policy/devicelist")
```
[the remark this answers is not in the document]
Comment (rtmooy) [reply catalystwan/endpoints/monitoring/security_policy.py:10]: Oh ok cool! I did not know we could do this, as there is no documentation or examples on this. Will make these changes

File: catalystwan/models/monitoring/security_policy.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2024 Cisco Systems, Inc. and its affiliates
+
+from pydantic import BaseModel, Field
+from typing import List
+
+
+class SecurityPolicyDeviceList(BaseModel):
+    #data: dict = Field(..., alias="data")
```
Comment (JimOverholt) [inline catalystwan/models/monitoring/security_policy.py:8]:
```suggestion
```

File: catalystwan/models/monitoring/security_policy.py (hunk 1 above)
Comment (rtmooy) [reply catalystwan/models/monitoring/security_policy.py]: Have made these changes

File: catalystwan/endpoints/monitoring/security_policy.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rtmooy) [reply catalystwan/endpoints/monitoring/security_policy.py]: Have made these changes, but not using DataSequence

File: catalystwan/models/monitoring/security_policy.py (hunk 3 above)
Comment (rtmooy) [reply catalystwan/models/monitoring/security_policy.py]: I have removed the line

**Review by JimOverholt — APPROVED** [review]

**Review by JimOverholt — APPROVED** [review]

**Review by sbasan — APPROVED** [review]

**Review by sbasan — APPROVED** [review]

**Pull request merged**